### DISCUSSION TITLE
TRACK AN SPECIFIC POINT IN A 2D DOMAIN

### DISCUSSION THREAD

**Discussion by FernandoIGuerra** [AMReX-Combustion/PeleC#754] [Q&A]
Hi, Everyone!, Is it possible to track a fixed point in the domain and print its values, such us velocity, kinetic energy and so on, at every time step? My simulation is a 2D geometry. Thanks in advance.

**Comment by baperry2** [comment]
You can probably accomplish this using runtime diagnostics capability, although it wasn't designed for this purpose. Use the `DiagConditional` type (intended for taking conditional averages), set `nbins` to 1, and apply `value_inrange` filters on `x` and `y` to capture only the cell you are interested in. 

https://amrex-combustion.github.io/PeleC/InputFiles.html#diagnostic-output

**Reply by FernandoIGuerra** [reply]
How may change the results the number of bins and the conditioning field? Also, the resulting average is calculated based on the defined region, filter on x and y? 
pelec.condT.nBins = 50                          
pelec.condT.condition_field_name = density

**Reply by baperry2** [reply]
Remember, this is a hack to get the data you want, so it involves doing weird things. In normal operation, (nBins > 1, filter covering a larger region), you get conditional averages - the average values across all cells for which the variable specified as condition_field_name falls the bin range for each bin.

With a nbins=1, which variable you choose as the conditioning variable shouldn't matter. Every cell that meets the filter critieria (i.e. both x and y coordinate in the ranges you specify) will simply be averaged together. Set the x and y coordinate filters so they only contain a single cell, and you should effectively just get the variable values in that cell saved to a file. You should get a different file generated at each timestep. Haven't tested this usage, but I think it should do what you want.

**Reply by Pandakii** [reply]
Hi @baperry2 , is it possible to write the output data from different timestep in a same file?

**Reply by baperry2** [reply]
This is not possible right now. I thought about adding this a while ago, but decided that it's easy enough to write a python script to combine data from multiple files for conditional diagnostics, so I didn't end up implementing that. There should be enough information to appropriately combine data from the Conditional diagnostic written across multiple timesteps into time-averaged conditional statistics.

**Comment by drummerdoc** [comment]
Check with AMReX for something called "station data". We used to support that long ago, but I'm not sure if it's been carried forward.

**Reply by marchdf** [reply]
That sounds useful. But I don't see a mention of it in amrex. Only a mention in a migration script.

**Reply by drummerdoc** [reply]
shoot!  It was a tool we used a lot back in the CPU days, and in particular when creating simulations of measured experiments.  I recall things getting complicated when we added MPI (that how long ago...), and I suspect things got too complex when dealing with data on devices.  However, then things might have gotten simpler again after years of GPU'izing codes and adding AMReX infrastructure to simplify moving things around.  All this is to say that I'll bet there are some AMReX underlying tools to make recreating StationData, or at least a one-off version of it, fairly trivial.  I'll ping the folks over there and report back with any pointers.

**Reply by drummerdoc** [reply]
Rumor has it that there may be something like that in AMReX in the next week...
